### PR TITLE
added test and coverage related targets

### PR DESCRIPTION
**Pull request by nurali-techie**
Fix: https://jira.coreos.com/browse/CRT-319

### CODE REVIEW

File: make/test.mk [hunk 1]
```diff
@@ -1,9 +1,117 @@
...
+## runs the tests with coverage
+test-with-coverage: generate
+	@echo "running the tests with coverage..."
 	@-mkdir -p $(COV_DIR)
-	@-rm -f $(COV_DIR)/coverage.txt
-	go test -timeout 10m -coverprofile=coverage.txt -covermode=atomic -coverpkg=$$GOPKGS ./...
+	@-rm $(COV_DIR)/coverage.txt
+	go test -timeout 10m -vet off ${V_FLAG} -coverprofile=$(COV_DIR)/coverage.txt -covermode=atomic ./...
+
+.PHONY: upload-codecov-report
```
Comment (alexeykazakov) [inline make/test.mk]: Let's update travis too so it doesn't publish coverage reports anymore - https://github.com/codeready-toolchain/registration-service/blob/master/.travis.yml#L43

Comment (nurali-techie) [reply make/test.mk]: Done https://github.com/codeready-toolchain/registration-service/pull/51/commits/06c64dc9ec0cc87be3fd389cdc86f24c994b0588

**Comment by nurali-techie** [comment]
@alexeykazakov thanks for quick review :-)

File: make/test.mk [hunk 2]
```diff
@@ -1,9 +1,117 @@
...
+test-e2e-local:
+	$(MAKE) test-e2e E2E_REPO_PATH=../toolchain-e2e
+
+.PHONY: test-e2e
+test-e2e: get-e2e-repo
+	# run the e2e test via toolchain-e2e repo
+ifeq ($(OPENSHIFT_BUILD_NAMESPACE),)
+	$(MAKE) -C ${E2E_REPO_PATH} test-e2e REG_REPO_PATH=${PWD}
+else
+	$(MAKE) -C ${E2E_REPO_PATH} test-e2e REG_REPO_PATH=${PWD} REG_IMAGE_NAME=registry.svc.ci.openshihost-operatorft.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:registration-service
```
[the remark this answers is not in the document]
Comment (nurali-techie) [reply make/test.mk]: Fixed: https://github.com/codeready-toolchain/registration-service/commit/144a3681082ef5cac551109d696fc632ae1eba8c

**Pull request merged**